### PR TITLE
Update devbox setup docs for x64 builds with newer Visual Studio versions

### PR DESCRIPTION
**Pull request by vaavva**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-c/blob/main/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `main` branch. 
  - [x] I have merged the latest `main` branch prior to submission and re-merged as needed after I took any feedback.
  - [x] I have squashed my changes into one with a clear description of the change.

# Reference/Link to the issue solved with this PR (if any)

# Description of the problem
build system tag needs to be specified outside of visual studio version for newer versions

# Description of the solution

### CODE REVIEW

**Review by danewalton — APPROVED** [review]

File: doc/devbox_setup.md [hunk 1]
```diff
@@ -71,7 +71,7 @@ cd cmake
   cmake .. -G "Visual Studio 16 2019" -A Win32
 ```
 
-> This builds x86 libraries. To build for x64 for Visual Studio 2017, `cmake .. -G "Visual Studio 15 2017 Win64"` or for Visual Studio 2019, `cmake .. -G "Visual Studio 16 2019 -A x64"`
+> This builds x86 libraries. To build for x64 for Visual Studio 2017, `cmake .. -G "Visual Studio 15 2017 Win64"` or for Visual Studio 2019, `cmake .. -G "Visual Studio 16 2019" -A x64`
```
Comment (CIPop) [inline doc/devbox_setup.md:74]: Please add VS 2022 as well. Here's the generator list from cmake: 

Verified that `cmake .. -G "Visual Studio 17 2022" -A x64` works on my devbox.

```
The following generators are available on this platform (* marks default):
* Visual Studio 17 2022        = Generates Visual Studio 2022 project files.
                                 Use -A option to specify architecture.
  Visual Studio 16 2019        = Generates Visual Studio 2019 project files.
                                 Use -A option to specify architecture.
  Visual Studio 15 2017 [arch] = Generates Visual Studio 2017 project files.
                                 Optional [arch] can be "Win64" or "ARM".
```

**Review by CIPop — APPROVED** [review]

File: doc/devbox_setup.md (hunk 1 above)
Comment (ericwol-msft) [reply doc/devbox_setup.md]: Please note, our 1ES images still build on 2017.

File: doc/devbox_setup.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ Build the sample project.
 
 In some cases, you may want to build the SDK locally for development and testing purposes (without using vcpkg). First, take the following steps to generate project files:
 
-- Open a "Developer Command Prompt for VS2017" or "Developer Command Prompt for VS2019".
+- Open a "Developer Command Prompt for VS2017" or "Developer Command Prompt for VS2019" or "Developer Command Prompt for VS 2022".
```
Comment (ericwol-msft) [inline doc/devbox_setup.md:60]: I would reorder this so that 2022 is first

File: doc/devbox_setup.md (hunk 1 above)
Comment (vaavva) [reply doc/devbox_setup.md]: > Please note, our 1ES images still build on 2017.

Is there anything I should specify about that in the doc?

**Pull request merged**